compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -542,7 +548,14 @@
 
         /* the body can be null in the case of Object type function declarations */
         if (funcNode.body != null) {
+
+            DefaultValueState prevDefaultValueState = this.defaultValueState;
+            if (defaultValueState == DefaultValueState.RECORD_FIELD_DEFAULT ||
```

Comment:
shall we use `this.defaultValueState` in these 2 intances too, looks like `this.` is used through out the function